src/lib.rs

```diff
@@ -563,3 +584,145 @@ impl Encoder for URIReserved {
         }
     }
 }
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum IriSegmentKind {
+    Segment,
+    SegmentNoColons,
+    Fragment,
+    Query,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct IriReserved(IriSegmentKind);
```

Comment:
I'm kind of annoyed by myself here. The name `IriReserved` is not consistent with `URIReserved`. However it is consistent with all the types names in my [`iref`](https://github.com/timothee-haudebourg/iref) crate... So I think we should keep yours and change `URIReserved` later on.

Reply:
We can add a type alias to the old name and mark it deprecated, and rename to UriReserved, without breaking semver.

Reply:
This won't work as you can't use a type alias in a constructor position (ie, in a function call), heh